sympy/polys/polytools.py

```diff
@@ -5974,8 +5970,11 @@ def _generic_factor_list(expr, gens, args, method):
 
     expr = sympify(expr)
 
-    if isinstance(expr, Expr) and not expr.is_Relational:
-        numer, denom = together(expr).as_numer_denom()
+    if isinstance(expr, (Expr, Poly)) and not expr.is_Relational:
```

Comment:
It looks like there is no need for `not expr.is_Relational`.

Reply:
I've removed this `is_Relational` check and another from nearby.